### PR TITLE
Revert "Revert "Have case upload set task result synchronously on success""

### PR DESCRIPTION
**Pull request by dannyroberts**
Reverts dimagi/commcare-hq#27273 and then incorporates https://github.com/dimagi/commcare-hq/pull/27289

### CODE REVIEW

**Review by millerdev — APPROVED** [review]

**Comment by dannyroberts** [comment]
A lot of build jobs were failing this afternoon (due to what appears to be network connection issue accessing github URLs), but one of the failures on the last build was real. Last PR should fix that

File: corehq/apps/case_importer/tests/test_importer.py [hunk 1]
```diff
@@ -69,7 +71,9 @@ def _config(self, col_names, search_column=None, case_type=None,
     @run_with_all_backends
     @patch('corehq.apps.case_importer.tasks.bulk_import_async.update_state')
     def testImportNone(self, update_state):
-        res = bulk_import_async.delay(self._config(['anything']), self.domain, None)
+        case_upload = CaseUploadRecord(upload_id=str(uuid.uuid4()), task_id=str(uuid.uuid4()))
+        case_upload.save()
+        res = bulk_import_async.delay(self._config(['anything']), self.domain, case_upload.upload_id)
```
Comment (millerdev) [inline corehq/apps/case_importer/tests/test_importer.py:76]: The name of this test is `testImportNone`, and the only `None` I can see in this test has been changed to `case_upload.upload_id`. Did the meaning of the test change? If yes, should it be renamed? Was the old test a nonsensical test? If not, maybe a new test should have been added instead of changing this one?

**Comment by dannyroberts** [comment]
Thanks for the thought here. Was hoping to get to this today, but instead I
just wanted to leave a note here that it's on my list of things to look
into tomorrow. I _think_ the test is still valid as it is and that None
refers to non-existent excel file, but I'm going to dig into the history to
see if I can pull up the original motivation and validate my reasoning

File: corehq/apps/case_importer/tests/test_importer.py (hunk 1 above)
Comment (dannyroberts) [reply corehq/apps/case_importer/tests/test_importer.py:76]: OK I finally took a deeper look here, and indeed, the "None" this was originally talking about disappeared from this test years ago: https://github.com/dimagi/commcare-hq/commit/f87f90439786a31c5905d43f784670212f2bc7e5#diff-916ed1d9c63ce6b7400d613eff5f64cfR68-R71, where I believe the first argument to `do_import` represents either the spreadsheet or the path to it. Now (and before I merged this PR) the same error is triggered inside `bulk_import_async` here https://github.com/dimagi/commcare-hq/blob/628c2105cf2605bd4bf573e4c1a2a028bf3d600c/corehq/apps/case_importer/tasks.py#L28 when it calls `case_upload.check_file()`, which raises a `ImporterRefError`, which converted to the string we check for in this test here: https://github.com/dimagi/commcare-hq/blob/8ab9a4dcd472eb9ddde4fc6ca6a93c21bd926cb2/corehq/apps/case_importer/util.py#L197-L202.

So in summary, I think the test has basically managed to test the same/equivalent thing as it and the code it tested morphed, and now the name of the test doesn't make all that much sense anymore. I'll change it to `testImportFileMissing`

Comment (dannyroberts) [reply corehq/apps/case_importer/tests/test_importer.py:76]: https://github.com/dimagi/commcare-hq/pull/27337
